Resources/Private/JavaScript/Host/Redux/UI/Inspector/index.js

```diff
@@ -0,0 +1,46 @@
+import {createAction} from 'redux-actions';
+import {$set, $drop} from 'plow-js';
+
+const WRITE_VALUE = '@packagefactory/guevara/UI/RightSidebar/WRITE_VALUE';
+const APPLY = '@packagefactory/guevara/UI/RightSidebar/APPLY';
+const APPLY_FINISHED = '@packagefactory/guevara/UI/RightSidebar/APPLY_FINISHED';
+const CANCEL = '@packagefactory/guevara/UI/RightSidebar/CANCEL';
+
+const writeValue = createAction(WRITE_VALUE, (nodeContextPath, propertyId, value) => ({nodeContextPath, propertyId, value}));
+const apply = createAction(APPLY, (nodeContextPath) => ({nodeContextPath}));
+const applyFinished = createAction(APPLY_FINISHED, (nodeContextPath) => ({nodeContextPath}));
+const cancel = createAction(CANCEL, (nodeContextPath) => ({nodeContextPath}));
+
+//
+// Export the actions
+//
+export const actions = {
+    writeValue,
+    apply,
+    applyFinished,
+    cancel
+
+};
+
+export const actionTypes = {
+    APPLY
+};
+
+//
+// Export the initial state
+//
+export const initialState = {
+    valuesByNodePath: {}
+};
+
+//
+// Export the reducer
+//
+export const reducer = {
+    [WRITE_VALUE]: ({nodeContextPath, propertyId, value}) => $set(['ui', 'inspector', 'valuesByNodePath', nodeContextPath, propertyId], value),
+    // APPLY is handled by a saga.
+    // APPLY_FINISHED is only triggered by the saga which listens on APPLY.
+    [APPLY_FINISHED]: ({nodeContextPath}) => $drop(['ui', 'inspector', 'valuesByNodePath', nodeContextPath]),
+    // TODO: APPLY_FINISHED needs to UPDATE the Node properties on the client. Or should the "Changes" API do this? I guess yes.
```

Comment:
Also it would be great to apply updates to the document without reloading the page, like we do currently in Ember UI: as feedback from the server, we could get the re-rendered markup for the updated node. I hate page reloads.